python/ray/data/read_api.py

```diff
@@ -277,6 +277,8 @@ def remote_read(i: int, task: ReadTask, stats_actor) -> MaybeBlockPartition:
         metadata.append(task.get_metadata())
 
     block_list = LazyBlockList(calls, metadata)
+    # XXX(ekl)
+    block_list._read_tasks = read_tasks
```

Comment:
TODO: maybe LazyBlockList should just take in a list of read tasks as the constructor.